frame/pablo/src/lib.rs

```diff
@@ -591,6 +600,18 @@ pub mod pallet {
 				);
 				if result.is_ok() {
 					weight += 1;
+					if let Some(updated_twap) = TWAPState::<T>::get(pool_id) {
```

Comment:
Its per pool , do you mean a separate event with Vec<TWAPUpdated> ?

Reply:
Not sure how this can help? Is it efficient for network layer , for runtime to emit event, or just because it is in on_initialize()? Is there any document from substrate for this kind of aggrigation? Also in update_twap() it is any way for a single pool so there I can't aggregate it. ( We can use event with a Vec of len 1) But I want to be sure that there is a real concern.

Reply:
I don't have a source for substrate other than general understanding of event based systems and batching, when you have a single event,
1. The network overhead is lower as there are fewer packets to be received and encoding/decoding load is lower
2. Any storage system only needs to store one set of event meta data instead of many, so efficient to store

But there could be advantages to not batching as well
1. Receivers might be able to process the msgs in a parallel/concurrent way
2. Receivers with lower resources can manage smaller events easily

I just saw a recent thread about this here https://substrate.stackexchange.com/questions/3455/emit-events-from-hooks-like-on-initialize, may be we can upvote/add detail

Reply:
:sweat_smile: I created that thread in hope to get some insights

Reply:
Cool, I didn't know that you are a "llvm ninja" :laughing:

Reply:
Let's keep an eye on the stackexchange thread and change if there is a authoritative answer for this